packages/block-library/src/calendar/index.php

```diff
@@ -69,6 +75,66 @@ function register_block_core_calendar() {
 
 add_action( 'init', 'register_block_core_calendar' );
 
+/**
+ * Builds an array of inline styles for the calendar block.
+ *
+ * @param  array $attributes The block attributes.
+ *
+ * @return string Style HTML attribute.
+ */
+function styles_for_block_core_calendar( $attributes ) {
+	$table_styles = array();
+
+	// Add color styles.
+	$has_text_color = ! empty( $attributes['style']['color']['text'] );
+	if ( $has_text_color ) {
+		$table_styles[] = sprintf( 'color: %s;', esc_attr( $attributes['style']['color']['text'] ) );
+	}
+
+	$has_background_color = ! empty( $attributes['style']['color']['background'] );
+	if ( $has_background_color ) {
+		$table_styles[] = sprintf( 'background-color: %s;', esc_attr( $attributes['style']['color']['background'] ) );
+	}
+
+	return ! empty( $table_styles ) ? sprintf( 'style="%s"', safecss_filter_attr( implode( ' ', $table_styles ) ) ) : '';
+}
+
+/**
+ * Returns color classnames depending on whether there are named or custom text and background colors.
+ *
+ * @param array $attributes The block attributes.
+ *
+ * @return string The color classnames to be applied to the block elements.
+ */
+function get_color_classes_for_block_core_calendar( $attributes ) {
```

Comment:
We might be able to reduce some of the duplicate logic in class generation if we leverage the style engine.